### PR TITLE
Create a TypedString type for values that are more than strings, but barely

### PR DESCRIPTION
**Pull request by alexwlchan**
This is spun out of a comment on https://github.com/wellcomecollection/catalogue-pipeline/pull/1607#discussion_r631735484

Is this behaviour useful? Yes, I think so.

Should it be copied in a dozen different places? Definitely not.

### CODE REVIEW

File: json/src/main/scala/weco/json/TypedString.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+trait TypedString[T <: TypedString[_]] {
+  val underlying: String
+
+  override def toString: String = underlying
+
+  def canEqual(a: Any): Boolean = a.getClass == this.getClass
+
+  override def equals(that: Any): Boolean =
+    that match {
+      case that if that.getClass == this.getClass =>
```
Comment (jamesgorrie) [inline json/src/main/scala/weco/json/TypedString.scala:51]: is `canEqual` what we should be using?

Comment (alexwlchan) [reply json/src/main/scala/weco/json/TypedString.scala]: Yup, might as well.

**Review by jamesgorrie — APPROVED** [review]
Smashing.

**Review by kenoir — COMMENTED** [review]
👏

**Pull request merged**